### PR TITLE
Atomic counter

### PR DESCRIPTION
**Pull request by michalbiesek**
Required:
- [x] #567

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/568)

### CODE REVIEW

File: test/memkind_memtier_test.cpp [hunk 1]
```diff
@@ -2,6 +2,7 @@
 /* Copyright (C) 2021 Intel Corporation. */
 
 #include <memkind_memtier.h>
+#include <thread>
```
Comment (PatKamin) [inline test/memkind_memtier_test.cpp:5]: nit: Add an extra blank line to split memkind header from standard library header.

File: test/memkind_memtier_test.cpp [hunk 2]
```diff
@@ -525,3 +526,35 @@ TEST_F(MemkindMemtierKindTest, test_tier_kind_check_size_posix_memalign_kind)
...
+        }));
+    }
+    for (size_t i = 0; i < num_threads; ++i) {
+        thds[i].join();
+    }
+
+    ASSERT_EQ(max_size, allocation_sum());
+
+    for (auto const &ptr : alloc_vec) {
+        memtier_free(ptr);
```
Comment (PatKamin) [inline test/memkind_memtier_test.cpp:558]: Could you test `memkind_atomic_decrement()` when using multiple threads, too?

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/568)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @PatKamin)

___
*[test/memkind_memtier_test.cpp, line 5 at r2](https://reviewable.io/reviews/memkind/memkind/568#-MW9GvgZRIYcIr-ZaCYj-r2-5:-MWT2YrvBsQsWjJ-aa5e:bb6un1j) ([raw file](https://github.com/memkind/memkind/blob/d402b4ed623f46d2edc786fec884251c1236b7fa/test/memkind_memtier_test.cpp#L5)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: Add an extra blank line to split memkind header from standard library header.
</blockquote></details>

Done
___
*[test/memkind_memtier_test.cpp, line 558 at r2](https://reviewable.io/reviews/memkind/memkind/568#-MW9GvgZRIYcIr-ZaCYj-r2-558:-MWT2Zm_9Fz7HPgysDYf:btqqj2v) ([raw file](https://github.com/memkind/memkind/blob/d402b4ed623f46d2edc786fec884251c1236b7fa/test/memkind_memtier_test.cpp#L558)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Could you test `memkind_atomic_decrement()` when using multiple threads, too?
</blockquote></details>

Done via free

**Review by PatKamin — COMMENTED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/568#-:-MWT8oxjDUq8-tcLeiP_:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @michalbiesek)

**Pull request merged**